### ISSUE TITLE
Pull export with  BibLaTeX remote statement returns HTTP code:500 in latex build

### ISSUE THREAD

**Issue by gtoubekis** [retorquere/zotero-better-bibtex#693]
Error Code EHFCJ2T4
Hi,
I try to include references via  BibLaTeX remote statement from a collection in Zotero
I use the url the I can retrieve via right-click on the collection and edit my  .tex file as follows.
_\addbibresource[location=remote]{http://localhost:23119/better-bibtex/collection?/0/0_0_test.biblatex}_
but receive an error - from the .blg I read this:
[1744] Utils.pm:180> ERROR - Could not fetch 'http://localhost:23119/better-bibtex/collection?/0/0_0_test.biblatex' (HTTP code: 500)
Any idea why the Internal Server error is produced?
 I checked in both browser I have, they show the reference correctly.

Thank in advance for the prompt reply.

Best 
Georgios

**Comment by retorquere** [comment]
There's not information in the current log. Can you turn on extended debugging and submit a new report?

**Comment by gtoubekis** [comment]
New ID PV7P6GN4
There was an issue reported with the developer of biber but with lower versions of biber, but I am running latest version of Miktex with  biber 2.7/biblatex 3.7 .

**Comment by retorquere** [comment]
Wait, if you open [http://localhost:23119/better-bibtex/collection?/0/0_0_test.biblatex]('http://localhost:23119/better-bibtex/collection?/0/0_0_test.biblatex) in your browser it shows the references correctly?

**Comment by gtoubekis** [comment]
yes. absolutely as it should be..

**Comment by retorquere** [comment]
And the debug log, was that generated by browser or by biber calling? I don't see anything relevant in the log file.

**Comment by gtoubekis** [comment]
The ERROR - Could not fetch 'http://localhost:23119/better-bibtex/collection?/0/0_0_test.biblatex' (HTTP code: 500)

comes from the biber /biblatex log. Seems that biber cannot access. I tried both .biblatex and bibtex format in the url. They both work fine in the browser but no chance to get it running in my latex build.

**Comment by gtoubekis** [comment]
I will contact the biber developer on that issue, maybe they have a clue.

**Comment by retorquere** [comment]
The alternative would be to install something like [fiddler](http://www.telerik.com/fiddler) and capture what's being sent to BBT -- but I see you're on Windows, and I'm not familiar with Windows, so I would be of little help getting it set up.

**Comment by gtoubekis** [comment]
I just tried out the push export  - works brilliant ! thanks for such a sophisticated piece of work.

**Comment by retorquere** [comment]
Does that mean you're not going to use pull export right now? I'll need your help to find out why you're getting the 500 error -- if you're going to use push export instead we'll have to close this issue for now. You can feel free to re-open or open a new issue whenever you want, but all my efforts except immediate help are going towards the Zoter0 5 rewrite right now.

**Comment by gtoubekis** [comment]
I tried to find out if biber connects at all to the webserver - used RawCap to monitor packages traffic on localhost - no experience with package monitoring - but I realized that while the pull request shows through the brwoser produced traffic and  output that can somehow be  guessed  viewing it with simple editor - there is no traffic i.e. no HTTP request reaching localhost from the \addbibressource call. I could not find out if there is a logfile of the embedded zotero webserver anywhere.
For the moment I assume the problem is on my side with either MikTex installation or misconfigured firewall settings.  The push function works fine with me, so I suggest to close the the issue for the time being.
